lib/ansible/plugins/action/template.py

```diff
@@ -139,7 +147,7 @@ def run(self, tmp=None, task_vars=None):
 
             old_vars = self._templar._available_variables
             self._templar.set_available_variables(temp_vars)
-            resultant = self._templar.do_template(template_data, preserve_trailing_newlines=True, escape_backslashes=False)
+            resultant = self._templar.do_template(template_data, preserve_trailing_newlines=True, escape_backslashes=False).encode(encode)
```

Comment:
Actually, there is nothing within do_template function to handle encoding. Eventually, we can use to_text function.